ddtrace/contrib/pymemcache/client.py

```diff
@@ -14,18 +18,36 @@
 from ddtrace import Pin
 from ddtrace.compat import reraise
 from ddtrace.ext import memcached as memcachedx
+from ddtrace.ext import net
+
+log = logging.getLogger(__name__)
+
 
 # keep a reference to the original unpatched clients
 _Client = Client
 
 
+DEFAULT_SERVICE = os.environ.get("DATADOG_SERVICE_NAME") or memcachedx.SERVICE
+
+
 class WrappedClient(wrapt.ObjectProxy):
+    """Wrapper providing patched methods of a pymemcache Client.
+
+    Relevant connection information is obtained during initialization and
+    attached to each span.
+
+    Keys are tagged in spans for methods that act upon a key.
+    """
+
     def __init__(self, *args, **kwargs):
         c = _Client(*args, **kwargs)
         super(WrappedClient, self).__init__(c)
 
+        # tags to apply to each span generated by this client
+        tags = _get_address_tags(*args, **kwargs)
+
         # attach the pin onto this instance
-        p = Pin(service=memcachedx.SERVICE, app_type=memcachedx.TYPE)
+        p = Pin(DEFAULT_SERVICE, tags=tags, app_type=memcachedx.TYPE)
```

Comment:
If the `Pin` is attached to `pymemcache`, you can `Pin.get_from(pymemcache)` and then override it with `tags`. Note that a `Pin` is immutable so it's safe to get and override (it will create a copy all the time).